internal/config/loader.go

```diff
@@ -77,73 +148,63 @@ func LoadConfig(useRegistry bool, profile string, confDir string) (config *commo
-		err = fmt.Errorf("connection to registry could not be made: %v", err.Error())
-	}
-
-	return registryClient, err
-}
-
 func checkRegistryUp(config *common.Config) error {
 	registryUrl := common.BuildAddr(config.Registry.Host, strconv.Itoa(config.Registry.Port))
 	fmt.Println("Check registry is up...", registryUrl)
 	fails := 0
 	for fails < config.Registry.FailLimit {
-		// http.Get return error in case of wrong HTTP method or invalid URL
-		// so we need to check for invalid status.
-		response, err := netClient.Get(registryUrl + consulStatusPath)
-		if err != nil {
-			fmt.Println(err.Error())
-			time.Sleep(time.Second * time.Duration(config.Registry.FailWaitTime))
-			fails++
-			continue
-		}
-
-		if response.StatusCode >= 200 && response.StatusCode < 300 {
+		if RegistryClient.IsAlive() {
 			break
-		} else {
-			return errors.New("bad response from Consul service")
 		}
+
+		time.Sleep(time.Second * time.Duration(config.Registry.FailWaitTime))
+		fails++
 	}
+
 	if fails >= config.Registry.FailLimit {
-		return errors.New("can't get connection to Consul")
+		return errors.New("can't get connection to Registry")
 	}
 	return nil
 }
 
-func newConsulClient(serviceName string, config *common.Config) (*registry.ConsulClient, error) {
-	consulClient := &registry.ConsulClient{}
-	registryConfig := registry.RegistryConfig{
-		Address:        config.Registry.Host,
-		Port:           config.Registry.Port,
-		ServiceName:    serviceName,
-		ServiceAddress: config.Service.Host,
-		ServicePort:    config.Service.Port,
-		CheckAddress:   fmt.Sprintf("http://%v:%v%v", config.Service.Host, config.Service.Port, common.APIPingRoute),
-		CheckInterval:  config.Registry.CheckInterval,
+func ListenForConfigChanges() {
+	if RegistryClient == nil {
+		common.LoggingClient.Error("listenForConfigChanges() registry client not set")
+		return
 	}
-	err := consulClient.Init(registryConfig)
-	return consulClient, err
-}
+
+	common.LoggingClient.Info("listen for config changes from Registry")
+
+	errChannel := make(chan error )
+	updateChannel := make(chan interface{})
+
+	RegistryClient.WatchForChanges(updateChannel, errChannel, &common.WritableInfo{}, common.WritableKey)
+
+	signalChan := make(chan os.Signal)
+	signal.Notify(signalChan, syscall.SIGINT, syscall.SIGTERM)
+
+	for {
+		select {
+		case <-signalChan:
+			// Quietly and gracefully stop when SIGINT/SIGTERM received
+			return
+		case ex := <-errChannel:
+			common.LoggingClient.Error(ex.Error())
+		case raw, ok := <-updateChannel:
+			if ok {
+				actual, ok := raw.(*common.WritableInfo)
+				if !ok {
+					common.LoggingClient.Error("listenForConfigChanges() type check failed")
+				}
+				common.CurrentConfig.Writable = *actual
+				common.LoggingClient.Info("Writeable configuration has been updated. Setting log level to " + common.CurrentConfig.Writable.LogLevel)
+				common.LoggingClient.SetLogLevel(common.CurrentConfig.Writable.LogLevel)
+			} else {
+				return
+			}
+		}
+	}
+}
```

Comment:
please add an empty line in the end of the file

Reply:
Ok